setup: `make gcc` step fails when homebrew is not installed in /usr/local

Hi Stefan,

I've run through your instructions successfully on both my intel Macs using homebrew but I just tried on an M1 Mac and hit an issue configuring/building gcc.  It complains about gmp.h not being found.

I have checked homebrew and on both Intel and M1 it sports version 6.2.1 of GMP installed with no caveats or issues.

After hitting the wall on `gmake all`and trying some googling I saw some suggestions for running `contrib/download_prerequisites` which I tried.  Got past the `configure gcc` step and into `make gcc`  at which point I hit a similar error "configuring in mpfr", which also complained about gmp.

I'm afraid I'm a bit new to this kind of GNU wizardry so may not be much help in answering further questions, but happy to try or send you the logs I have.

As I say, homebrew seemed happy on both platforms and all worked fine on my intel Macs.

ok last investigation.

I went into `projects/gcc` and it has a `gmp-4.3.2` subdir so I tried `./configure --with-gmp = ./gmp-4.3.2` which failed because (when I looked) there's no .h file there.

There is a `.h.in` so on a whim I tried to do a `./configure` inside that `gmp-4.3.2` folder and got this:

`configure: error: Oops, mp_limb_t is 64 bits, but the assembler code in this configuration expects 32 bits.`

So.... the gmp inside gcc can't be configured, which is why gcc is complaining. (I think).  I've gone as far as I can, I think so will leave it alone in case anyone can figure this out :-)

Guess I can't actually leave it alone.  I edited the makefile in amiga-gcc to call configure for GCC and explicitly pass /opt/homebrew for GMP, MPFR and MCP (because right now the makefile downloads those into projects/gcc instead of using the homebred versions) and I get a *lot* further.

Now I'm hitting this when building GCC (again, just leaving notes for any interested party):

Undefined symbols for architecture arm64:
  "_host_hooks", referenced from:
      __Z20c_common_no_more_pchv in c-pch.o
      __ZN6toplev4mainEiPPc in libbackend.a(toplev.o)
      __Z14gt_pch_restoreP7__sFILE in libbackend.a(ggc-common.o)
      __Z11gt_pch_saveP7__sFILE in libbackend.a(ggc-common.o)
ld: symbol(s) not found for architecture arm64
collect2: error: ld returned 1 exit status
gmake[2]: *** [/Users/test/work/amiga-gcc/projects/gcc/gcc/c/Make-lang.in:71: cc1] Error 1
make gcc...failed
use less "log/make gcc.log" to view the full log and search for ***
gmake: *** [Makefile:419: build-Darwin/gcc/_done] Error 2

--- a/Makefile
+++ b/Makefile
@@ -433,7 +433,10 @@ CONFIG_GCC = --prefix=$(PREFIX) --target=m68k-amigaos --enable-languages=c,c++,o
 
 # OSX : libs added by the command brew install gmp mpfr libmpc
 ifeq (Darwin, $(findstring Darwin, $(UNAME_S)))
-	CONFIG_GCC += --with-gmp-include=/usr/local/include --with-gmp-lib=/usr/local/lib --with-mpfr-include=/usr/local/include --with-mpfr-lib=/usr/local/lib --with-mpfr-include=/usr/local/include --with-mpfr-lib=/usr/local/lib
+	BREW_PREFIX := $$(brew --prefix)
+	CONFIG_GCC += --with-gmp=$(BREW_PREFIX) \
+		--with-mpfr=$(BREW_PREFIX) \
+		--with-mpc=$(BREW_PREFIX)
 endif
 
 

--- a/README.md
+++ b/README.md
@@ -27,32 +27,27 @@ If building with a normal user, the `PREFIX` directory must be writable (default
 ### macOS
 Install Homebrew (https://brew.sh/) or any other package manager first. The compiler will be installed together with XCode. Once XCode and Homebrew are up install the required packages:
 
-`brew install bash wget make lhasa gmp mpfr libmpc flex gettext gnu_sed texinfo gcc make autoconf`
+```
+brew install bash wget make lhasa gmp mpfr libmpc flex gettext gnu-sed texinfo gcc@11 make autoconf
+```
 
 By default macOS uses an outdated version of bash. Therefore, on macOS host always pass the the SHELL=/usr/local/bin/bash parameter (or any other valid path pointing to bash), e.g.:
+
 ```
-make all SHELL=/usr/local/bin/bash
+make all SHELL=$(brew --prefix)/bin/bash
 ```
+
 On macOS it may be also necessary to point to the brew version of gcc make and autoconf, e.g.:
+
 ```
-CC=gcc-9 CXX=g++-9 gmake all SHELL=/usr/local/bin/bash
+CC=gcc-11 CXX=g++-11 gmake all SHELL=$(brew --prefix)/bin/bash
 ```
 
 **ALSO NOTE** If you want `m68k-amigaos-gdb` then you have to build it with `gcc`
 
 ### macOs on Mi1
 
-Native build on M1 Mac has proved difficult at this stage for a number of reasons. I filed (and closed) a separate issue. 
-
-We can, however, get an x86 version of the toolchain working on M1 with these steps.
-
-1. Ensure rosetta is installed. `/usr/sbin/softwareupdate --install-rosetta --agree-to-license`
-2. Ensure Xcode command line tools are installed `xcode-select --install`
-3. Install convince Homebrew to install an x86_64 version instead `arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"`.  Ensure `/usr/local/bin` is early in your path.
-4. Use the x86 homebrew to install the required packages `arch -x86_64 brew install bash wget make lhasa gmp mpfr libmpc flex gettext texinfo gcc make autoconf`
-5. `git clone git@github.com:bebbo/amiga-gcc ; cd amiga-gcc`
-6. specify x86 when building `CC=gcc-10 CXX=g++-10 SHELL=/usr/local/bin/bash arch -x86_64 gmake clean update all`. 
-7. profit!
+Native builds on M1 Macs are now directly supported.
 
 ### Windows with Cygwin
 Install cygwin via setup.exe and add wget. Then open cygwin shell and run:

